packages/create-discord-bot/src/create-discord-bot.ts

```diff
@@ -1,70 +1,85 @@
-	process.exit(1);
+interface Options {
+	directory: string;
+	javascript?: boolean;
+	typescript?: boolean;
 }
 
-const root = path.resolve(directory);
-const directoryName = path.basename(root);
+export async function createDiscordBot({ typescript, javascript, directory }: Options) {
+	if (!directory) {
+		console.error(chalk.red('Please specify the project directory.'));
+		process.exit(1);
+	}
 
-if (existsSync(root) && readdirSync(root).length > 0) {
-	console.error(chalk.red(`The directory ${chalk.yellow(`"${directoryName}"`)} is not empty.`));
-	console.error(chalk.red(`Please specify an empty directory.`));
-	process.exit(1);
-}
+	const root = path.resolve(directory);
+	const directoryName = path.basename(root);
 
-// We'll use the directory name as the project name. Check npm name validity.
-const validationResult = validateProjectName(directoryName);
+	const directoryStats = await stat(root).catch(async (error) => {
+		// Create a new directory if the specified one does not exist.
+		if (error.code === 'ENOENT') {
+			await mkdir(root, { recursive: true });
+			return stat(root);
+		}
 
-if (!validationResult.validForNewPackages) {
-	console.error(
-		chalk.red(
-			`Cannot create a project named ${chalk.yellow(`"${directoryName}"`)} due to npm naming restrictions.\n\nErrors:`,
-		),
-	);
+		throw error;
+	});
 
-	for (const error of [...(validationResult.errors ?? []), ...(validationResult.warnings ?? [])]) {
-		console.error(chalk.red(`- ${error}`));
+	// If a directory exists and it's not empty, throw an error.
+	if (directoryStats.isDirectory() && (await readdir(root)).length > 0) {
+		console.error(chalk.red(`The directory ${chalk.yellow(`"${directoryName}"`)} is not empty.`));
+		console.error(chalk.red(`Please specify an empty directory.`));
+		process.exit(1);
 	}
 
-	console.error(chalk.red('\nSee https://docs.npmjs.com/cli/configuring-npm/package-json for more details.'));
-	process.exit(1);
-}
+	// We'll use the directory name as the project name. Check npm name validity.
+	const validationResult = validateProjectName(directoryName);
 
-if (!existsSync(root)) {
-	mkdirSync(root, { recursive: true });
-}
+	if (!validationResult.validForNewPackages) {
+		console.error(
+			chalk.red(
+				`Cannot create a project named ${chalk.yellow(
+					`"${directoryName}"`,
+				)} due to npm naming restrictions.\n\nErrors:`,
+			),
+		);
 
-console.log(`Creating ${directoryName} in ${chalk.green(root)}.`);
-cpSync(new URL(`../template/${typescript ? 'TypeScript' : 'JavaScript'}`, import.meta.url), root, { recursive: true });
+		for (const error of [...(validationResult.errors ?? []), ...(validationResult.warnings ?? [])]) {
+			console.error(chalk.red(`- ${error}`));
+		}
 
-process.chdir(root);
+		console.error(chalk.red('\nSee https://docs.npmjs.com/cli/configuring-npm/package-json for more details.'));
+		process.exit(1);
+	}
+
+	if (!directoryStats.isDirectory()) {
+		await mkdir(root, { recursive: true });
+	}
```

Comment:
```suggestion
```
This may be completely redundant too? Since something similar is already done prior to this.